src/app/(proper_react)/redesign/(authenticated)/user/dashboard/page.tsx

```diff
@@ -33,7 +40,45 @@ export default async function DashboardPage() {
     return redirect("/redesign/user/welcome/");
   }
 
+  // This contains the latest scan results in our database.
   const latestScan = await getLatestOnerepScanResults(profileId);
+
+  // Attempt to fetch the current scan results from the provider.
+  // If anything is newer, add it to the database.
+  // Continue if there are any errors.
+  try {
+    const currentScan = await getAllScanResults(profileId);
+    const newScanResults: ScanResult[] = [];
+    currentScan.forEach((current) => {
+      latestScan.results.forEach((latest) => {
```

Comment:
This might be easier to follow using something like `const newScanResults = currentScan.map(({ results }) => results.filter(/* ... */)).flat()`?

(But that might also just be personal preference.)

Reply:
Hm yeah I don't use the various array functions enough... I wonder if https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap could simplify it further? Will see what I can come up with.

Reply:
OK I can't figure out a way that seems more readable :/ to be clear, what this needs to do is:

1. fetch the latest results from remote and the db and add any new scan IDs
2. compare each element in the array, if any IDs match then only update if `updated_at` is newer

Reply:
Woah! I did not know that `.flatMap` was actually already standardised and widely supported! I'd have written a more complete diff, but it's merged already, so :shrug: